### PR TITLE
#160702239 Ft authentication 

### PR DESCRIPTION
**Pull request by danuluma**
#### What does this PR do?
Adds authentication using jwt

#### Description of Task to be completed?
adds the endpoints:
```~/dann/api/v1/reg``` --> here a new user can register to the app in order to get access to secured endpoints
```~/dann/api/v1/login``` --> a returning user logs in using his/her username and password to get access to secured endpoints

#### How should this be manually tested?
* Clone the repo.
* Run ```python3 run.py```
* Fire up postman and navigate to POST ```/dann/api/v1/reg```. Register to get an access token.
* Create another request to GET ```/dann/api/``` . It should return unauthorized access.
* Now add to the request header the access token gotten previously in the POST ```/dann/api/v1/reg```. It should return "Hello world". You will have successfully gained access using an authorization token.

#### What are the relevant pivotal tracker stories?
#160702239

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+instance/config.py
+.pytest_cache
+requirements2.txt
+*pycache*
+*pytest_cache*
```
Comment (mwaz) [inline .gitignore]: add a blank line here

File: .pytest_cache/v/cache/lastfailed [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  "app/tests/v1/conftest.py::test_get_order1": true, 
+  "app/tests/v1/conftest.py::test_home": true, 
+  "app/tests/v1/dann_test.py::test_create_order": true, 
+  "app/tests/v1/dann_test.py::test_edit_order": true, 
+  "app/tests/v1/dann_test.py::test_get_all": true, 
+  "app/tests/v1/dann_test.py::test_get_order1": true, 
+  "app/tests/v1/dann_test.py::test_home": true, 
+  "app/tests/v1/test_app.py": true, 
+  "app/tests/v1/test_views.py": true
+}
```
Comment (mwaz) [inline .pytest_cache/v/cache/lastfailed:12]: add the pytest_cache directory in your .gitignore

File: .travis.yml [hunk 3]
```diff
@@ -1,7 +1,17 @@
...
   - "3.6"
+
+
 install:
   - pip install -r requirements.txt
+  - pip install coveralls codecov
+
 script:
-  - python3 -m pytest
+  - python3 -m unittest
```
Comment (mwaz) [inline .travis.yml:12]: choose whether you want to run your tests solely using unittest or nosetest not both

File: .gitignore (hunk 1 above)
Comment (danuluma) [reply .gitignore]: Thanks, I've now added it

File: .pytest_cache/v/cache/lastfailed (hunk 2 above)
Comment (danuluma) [reply .pytest_cache/v/cache/lastfailed:12]: I've done it

File: .travis.yml (hunk 3 above)
Comment (danuluma) [reply .travis.yml:12]: I've chosen unittest only in the latest commit

**Pull request merged**